include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -298,6 +300,20 @@ namespace internal
             fe->shape_grad_grad(my_i, q_point)[0][0];
         }
 
+      quadrature_data_on_face[0].resize(quad.size() * 3);
+      quadrature_data_on_face[1].resize(quad.size() * 3);
+
+      dealii::FE_DGQArbitraryNodes<1> fe_quad(quad);
+
+      for (unsigned int i = 0; i < quad.size(); ++i)
+        {
+          Point<1> q_point;
+          q_point[0]                    = 0;
+          quadrature_data_on_face[0][i] = fe_quad.shape_value(i, q_point);
+          q_point[0]                    = 1;
+          quadrature_data_on_face[1][i] = fe_quad.shape_value(i, q_point);
+        }
```

Comment:
I think you need to encapsulate this part in some `if` for the case the number of quadrature points is very high, say a few hundreds. In that case, the `FE_DGQArbitraryNodes` polynomials would underflow and result in a division by zero. The collocation is not really helpful in that case anyway.